### PR TITLE
Feature: Award Data Archive delta files (created and modified only)

### PR DESCRIPTION
**Pull request by michaeldhess**
**High level description:**
Script that generates delta files for data that has changed since the last monthly file generation.

**Technical details:**
New script, `download/management/commands/populate_monthly_delta_files.py`, which generates a file of Contract or Assistance records that have been changed or created since the `last_date` parameter. The script utilizes functions from the regular download endpoints, but does not use the endpoints themselves because it must be customized. 

**Link to JIRA Ticket:**
[DEV-910](https://federal-spending-transparency.atlassian.net/browse/DEV-910)

**The following are ALL required for the PR to be merged:**
- [ ] Backend review completed
- Matview impact assessment completed (N/A)
- Frontend impact assessment completed (N/A)
- [ ] Data validation completed (In Progress)
- API Performance evaluation completed and present (N/A)

### CODE REVIEW

File: usaspending_api/download/management/commands/populate_monthly_delta_files.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    def download(self, award_type, agency='all', generate_since=None):
+        """Create a delta file based on award_type, and agency_code (or all agencies)"""
+        award_map = AWARD_MAPPINGS[award_type]
+
+        # Create Source and update fields to include correction_delete_ind
+        source = CsvSource('transaction', award_map['letter_name'].lower(), 'transactions')
+        source.query_paths.update({
+            'correction_delete_ind': award_map['correction_delete_ind']
+        })
+        source.query_paths.move_to_end('correction_delete_ind', last=False)
```
Comment (dpb-bah) [inline usaspending_api/download/management/commands/populate_monthly_delta_files.py]: +1 for `move_to_end`, never encountered it before

File: usaspending_api/download/management/commands/populate_monthly_delta_files.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        ordered_map = OrderedDict(zip(aliases, selects))
+        selects_str = ", ".join(cases + ['{} AS \"{}\"'.format(select, alias) for alias, select in ordered_map.items()])
+
+        return raw_query.replace(select_string, selects_str)
+
+    def parse_filters(self, award_types, agency):
+        """Convert readable filters to a filter object usable for the matview filter"""
+        award_type_codes = []
+        for award_type in award_types:
+            award_type_codes = award_type_codes + all_award_types_mappings[award_type]
```
Comment (dpb-bah) [inline usaspending_api/download/management/commands/populate_monthly_delta_files.py]: More of a code style preference, your call:
` award_type_codes = [all_award_types_mappings[award_type] for award_type in award_types]`

File: usaspending_api/download/management/commands/populate_monthly_delta_files.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+            'correction_delete_ind': award_map['correction_delete_ind']
+        })
+        source.query_paths.move_to_end('correction_delete_ind', last=False)
+        source.human_names = list(source.query_paths.keys())
+
+        # Apply filters to the queryset
+        filters, agency_code = self.parse_filters(award_map['award_types'], agency)
+        source.queryset = VALUE_MAPPINGS['transactions']['filter_function'](filters)
+        if award_type == 'Contracts':
+            # Derive the correction_delete_ind from the created_at of the records
```
Comment (dpb-bah) [inline usaspending_api/download/management/commands/populate_monthly_delta_files.py]: Stupid question: why is this being applied to contracts? I thought both the `correction_delete_ind` were being added via line 50-52?

File: usaspending_api/download/management/commands/populate_monthly_delta_files.py [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+            source.queryset = source.queryset. \
+                annotate(correction_delete_ind=Case(When(transaction__contract_data__created_at__gte=generate_since,
+                                                    then=Value('C')), default=Value(''), output_field=CharField()))
+        source.queryset = source.queryset.filter(**{
+            'transaction__{}__{}__gte'.format(award_map['model'], award_map['date_filter']): generate_since
+        })
+        source_query = source.row_emitter(None)
```
Comment (dpb-bah) [inline usaspending_api/download/management/commands/populate_monthly_delta_files.py]: Can we add a comment here explaining None represents all columns?

File: usaspending_api/download/management/commands/populate_monthly_delta_files.py (hunk 3 above)
Comment (michaeldhess) [reply usaspending_api/download/management/commands/populate_monthly_delta_files.py]: The `correction_delete_ind` in the `AWARD_MAPPINGS` represents the mapping from query to file header, with the key being the file-level header and the value being the column name within the query.

The code here defines the data within the query in order for it to be mapped to a file header in the first place.

**Pull request merged**